libs/@guardian/libs/src/storage/storage.ts

```diff
@@ -98,6 +98,38 @@ class StorageFactory {
 	setRaw(key: string, value: string): void {
 		return this.#storage?.setItem(key, value);
 	}
+
+	/**
+	 * Get all of the keys in storage.
+	 *
+	 * @returns an array of all keys in the storage object,
+	 * or undefined if storage is not available.
+	 *
+	 * @note This is not a method in the native API, and gets
+	 * all keys by iterating over the entire storage object.
+	 * It is therefore potentially subject to race conditions
+	 * and if other code is modifying storage at the same time
+	 * then the results may be inaccurate.
+	 *
+	 * @note This method depends on the `key` method on the native API,
+	 * which gets keys by index. But the _order_ of keys in storage is
+	 * dependent on user-agent implementation, and so you should not rely
+	 * on the order of keys returned by this method. (See
+	 * https://developer.mozilla.org/en-US/docs/Web/API/Storage/key for more.)
```

Comment:
```suggestion
```

as with my comment above, it might be useful to move this info to the part of the code where `key` is used. it's certainly useful for people working on the API in the future, but doesn't particularly help people who are using our method (what can they actually do with that info?)